src/endpoint/s3/s3_rest.js

```diff
@@ -96,6 +96,12 @@ async function handle_request(req, res) {
         error_token_expired: S3Error.ExpiredToken,
         auth_token: () => signature_utils.make_auth_token_from_request(req)
     };
+    // AWS s3 returns an empty response when s3 request sends without host header.
+    if (!req.headers.host) {
+        dbg.warn('s3_rest: handle_request: S3 request is missing host header, header ', req.headers);
+        res.end();
```

Comment:
Don't we need to return a status code?
I see in other examples that before we use `res.end(); return;` we set `res.statusCode = `.

Reply:
@shirady I can add Bad Request error code(400), WDYT?

Reply:
@naveenpaul1 it's not our decision, we need to follow AWS, make sure we return what they return

Reply:
@romayalon AWS is returning `307: Temporary Redirect` error code with a location 
`Location: https://aws.amazon.com/s3/`, Since we dont have a location to redirect can we leave the location empty?
```
curl https://{bucket-name}.s3.amazonaws.com --http1.0 -H 'Host:' -v
HTTP/1.1 307 Temporary Redirect
< x-amz-id-2: eQref0gQxYWqX3+wYaVPfalsHqkifllRfkuOiN2E5LlcTkI/xDphkfxuj4/xG4GJXlRY8lOPqUjCEa5OOJCn//JfmwUgf7vLvDr/U/kM92k=
< x-amz-request-id: H5WMCF0G2HM2P1J8
< Date: Tue, 17 Dec 2024 09:14:10 GMT
< Location: https://aws.amazon.com/s3/
< Content-Length: 0
< Server: AmazonS3
< Connection: close

```

Reply:
I think you need to rerun it again with the `-L` flag: `curl https://noobaa-core-rpms.s3.amazonaws.com --http1.0 -H 'Host:' -v -L` (the `noobaa-core-rpms` is a public bucket name, you can use `curl https://noobaa-core-rpms.s3.amazonaws.com --http1.0 -H 'Host:' -v -L | grep "HTTP/1.1"` for smaller output) and then compare the results - you should see that in AWS it starts with "HTTP/1.1 307 Temporary Redirect" and then "HTTP/1.1 200", and I think we return also 200 status code (`curl https://localhost:6443 --http1.0 -H 'Host:' --insecure -v` "HTTP/1.1 200 OK").
So I'm not sure we need to add the status code as I asked.

Reply:
@shirady  because of the redirected URL in this case its `https://aws.amazon.com/s3/` is a valid url and this redirected request return complete s3 aws web page with a response code 200.

Reply:
@naveenpaul1 I don't think this is enough, this is true to http1.0, I just noticed that when using HTTP1.1 we get  `HTTP/1.1 400 Bad Request`
`curl https://noobaa-core-rpms.s3.amazonaws.com -H 'Host:' -v`
`curl https://noobaa-core-rpms.s3.amazonaws.com -H 'Host:' -v -L`
please try to do some more research here about the proper behavior

Reply:
@romayalon checked AWS doc and Host header is required for AWS request, and